lib/X509/CSR/CSR.php

```diff
@@ -73,6 +75,23 @@ protected function createCSRSequence()
         $this->addChild($signature);
     }
 
+    public function getSignatureSubject()
+    {
+        $versionNr            = new Integer(self::CSR_VERSION_NR);
+        $publicKey            = new PublicKey($this->publicKey);
+
+        $certRequestInfo  = new Sequence($versionNr, $this->subject, $publicKey);
+        return $certRequestInfo->getBinary();
+    }
+
+    public function setSignature($signature, $signatureAlgorithm = OID::SHA1_WITH_RSA_SIGNATURE)
+    {
+        $this->signature = $signature;
+        $this->signatureAlgorithm = $signatureAlgorithm;
+
+        $this->createCSRSequence();
```

Comment:
I note createCSRSequence is non-public, perhaps setSignature should check if $this (extends Sequence) has any children yet?

https://github.com/fgrosse/PHPASN1/blob/73d914e2c570c1d25a8b3be1cc4a71347f2aca62/lib/X509/CSR/CSR.php#L73-L76

 - someone creates CSR() _with_ a signature and calls setSignature to override it
 - someone creates CSR() _without_ a signature, and somehow calls setSignature twice

Maybe it should error, or remove all the children before calling createCSRSequence, or maybe clone $this, remove children, call createCSRSequence so the original object remains valid?

Reply:
I opted for clearing the sequence inside createCSRSequence().